__tests__/app.test.js

```diff
@@ -0,0 +1,34 @@
+const app = require('../app');
+const seed = require('../db/seeds/seed');
+const data = require('../db/data/test-data/index');
+const connection = require('../db/connection');
+const request = require('supertest');
+const { forEach } = require('../db/data/test-data/articles');
+
+afterAll(() => {
+    return connection.end();
+})
+
+beforeEach(() => {
+    return seed(data);
+})
+
+describe("/api/topics", () => {
+    test("200: response with a status of 200", () => {
+        return request(app).get("/api/topics").expect(200)
+    });
+    test('"200: responds with an array of topic objects"', () => {
+        return request(app).get("/api/topics").expect(200)
+        .then(({ body }) => {
+            const topics = body.rows
+            expect(Array.isArray(topics)).toBe(true)
+
+            topics.forEach((topic) => {
+                expect(topic).toHaveProperty('slug')
+                expect(topic).toHaveProperty('description')
+            })
```

Comment:
One potential issue with this test is if your API returns an _empty array_ `[ ]` , the test will pass. This is because the `forEach` will skip if the array is empty. 

How can we something into this test to ensure that the array has the expected number of elements ( based on the test data ) ?